synapse/lib/platforms/linux.py

```diff
@@ -104,6 +105,25 @@ def getTotalMemory():
     logger.warning('Unable to find max memory limit')  # pragma: no cover
     return 0  # pragma: no cover
 
+
+def getSysctls():
+    _sysctls = (
+        ('vm.swappiness', '/proc/sys/vm/swappiness'),
+        ('vm.dirty_expire_centisecs', '/proc/sys/vm/dirty_expire_centisecs'),
+        ('vm.dirty_writeback_centisecs', '/proc/sys/vm/dirty_writeback_centisecs'),
+        ('vm.dirty_background_ratio', '/proc/sys/vm/dirty_background_ratio'),
+        ('vm.dirty_ratio', '/proc/sys/vm/dirty_ratio'),
+    )
+    ret = {}
+    for key, fp in _sysctls:
+        if os.path.isfile(fp):
+            with open(fp) as f:
+                valu = f.read().strip()
+                ret[key] = valu
```

Comment:
Are any of these values types that we should cast from strings in case we want to process them? Could we add a func callback to the list and norm them by calling int/float/etc?

Reply:
the normalization routines would def need to account for that. somebody has to do it to reason over the values and it makes more sense to do it closer to the origin

Reply:
per discussion we'll leave anything missing as `None` and norm the rest